### PR TITLE
Update ListAlertsUseCase to use new Tenure Information API

### PR DESCRIPTION
**Pull request by LBHCallumM**
## Link to JIRA ticket

[Story](https://www.pivotaltracker.com/story/show/180329680)

## Describe this PR

ListAlertsUseCase needed update to use new TenureInformation API.

#### _Checklist_

- [ ] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [ ] Added tests to cover all new production code
- [ ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [ ] Checked all code for possible refactoring
- [ ] Code pipeline builds correctly

### CODE REVIEW

**Review by teodor-hristoforov-hackney — APPROVED** [review]

File: RepairsApi/V2/UseCase/ListAlertsUseCase.cs [hunk 1]
```diff
@@ -12,23 +12,61 @@ public class ListAlertsUseCase : IListAlertsUseCase
...
+
+                _logger.LogInformation("Getting TenancyAgreementReference from new Tenure API");
+
+                return await _tenureInformationGateway.GetById(asset.Tenure.Id);
+            }
+
+            // old endpoint
+            _logger.LogInformation("Getting TenancyAgreementReference from old UH Tenure API");
+
+            return await _universalHousingTenancyGateway.GetTenancyInformationAsync(propertyReference);
```
Comment (teodor-hristoforov-hackney) [inline RepairsApi/V2/UseCase/ListAlertsUseCase.cs:69]: Just for my information, is this old api call be changed in a future branch?

Comment (LBHCallumM) [reply RepairsApi/V2/UseCase/ListAlertsUseCase.cs:69]: Yes, I will remove it later when feature toggling isn't required. There were a lot of changes, and I didn't want to block any other work.

Comment (teodor-hristoforov-hackney) [reply RepairsApi/V2/UseCase/ListAlertsUseCase.cs:69]: Thanks for letting me know.

File: RepairsApi/V2/UseCase/ListAlertsUseCase.cs [hunk 2]
```diff
@@ -12,23 +12,61 @@ public class ListAlertsUseCase : IListAlertsUseCase
...
+
             var personAlertList = await _alertsGateway.GetPersonAlertsAsync(tenureInformation?.TenancyAgreementReference);
             return new AlertList
             {
                 PropertyAlerts = propertyAlertList,
                 PersonAlerts = personAlertList
             };
         }
+
+        private async Task<TenureInformation?> GetTenureInformation(string propertyReference)
```
Comment (Asura5) [inline RepairsApi/V2/UseCase/ListAlertsUseCase.cs:51]: I saw this in another PR. Could we refactor out using the Adapter pattern? Reason: breaking DRY

File: RepairsApi/V2/UseCase/TenureInformationUseCase.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+#nullable enable
+        public async Task<TenureInformation?> GetTenureInformationWithPropertyReference(string? propertyReference)
+        {
+            if (propertyReference == null) return null;
+
+            if (await _featureManager.IsEnabledAsync(FeatureFlags.EnableNewTenureAPI) == true)
+            {
+                // new endpoint
+                _logger.LogInformation("Getting TenureId from Asset API");
```
Comment (Asura5) [inline RepairsApi/V2/UseCase/TenureInformationUseCase.cs:40]: Please add the ID to this log

File: RepairsApi/V2/UseCase/TenureInformationUseCase.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+                var asset = await _assetGateway.GetById(propertyReference);
+                if (asset == null) return null;
+
+                _logger.LogInformation("Getting TenancyInformation from new Tenure API");
+
+                return await _tenureInformationGateway.GetById(asset.Tenure.Id);
+            }
+
+            // old endpoint
+            _logger.LogInformation("Getting TenancyInformation from old UH Tenure API");
```
Comment (Asura5) [inline RepairsApi/V2/UseCase/TenureInformationUseCase.cs:51]: Here too

File: RepairsApi/V2/UseCase/TenureInformationUseCase.cs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+            if (await _featureManager.IsEnabledAsync(FeatureFlags.EnableNewTenureAPI) == true)
+            {
+                // new endpoint
+                _logger.LogInformation("Getting TenureId from Asset API");
+
+                var asset = await _assetGateway.GetById(propertyReference);
+                if (asset == null) return null;
+
+                _logger.LogInformation("Getting TenancyInformation from new Tenure API");
```
Comment (Asura5) [inline RepairsApi/V2/UseCase/TenureInformationUseCase.cs:45]: Here too

**Review by Asura5 — APPROVED** [review]

**Pull request merged**